### DISCUSSION TITLE
About the birt pdf report some fonts are not working

### DISCUSSION THREAD

**Discussion by Akshay-64** [eclipse-birt/birt#1236] [Q&A] [answered]
Hi everyone
Hindi , Bengali  characters are not displaying in generated pdf in birt report. I made the changes in fontConfig.xml by the adding the font  NotoSansDevanagari for the hindi but in generated pdf it was showing empty characters. Can anyone please help in this.

**Comment by speckyspooky** [comment]
Could you please provide the link to download your font version because I found many different of it.
And perhaps it would help a screen of your designer to see the font there with example labels and also a version of your PDF output to see the empty characters.

**Comment by Akshay-64** [comment]
https://fonts.google.com/noto/specimen/Noto+Sans+Devanagari 
above was the font link for hindi language. I am using birt 4.11 version.

**Comment by hvbtup** [comment]
Which of the several *.ttf fonts did you add to fontsConfig.xml?
Can you provide your fontsConfig.xml and a stand-alone report demonstrating the issue?

**Reply by Akshay-64** [reply]
[font.txt](https://github.com/eclipse/birt/files/11027340/font.txt)
https://fonts.google.com/noto/specimen/Noto+Sans+Devanagari
[fda.pdf](https://github.com/eclipse/birt/files/11027372/fda.pdf)
![image](https://user-images.githubusercontent.com/65806980/226588399-1ae8bc13-d977-4b00-9b83-cdc60b32c709.png)

In above I added fontconfig in text format because git doesn't supporting to upload xml format 
in html format hindi text was able to display while in pdf unable to display the hindi text

**Comment by speckyspooky** [comment, marked as answer]
So I have verified it on my laptop.
Step 01: Installed the "Noto Sans Devanagari Regular.ttf".
Step 02: Afterwards I added the installation path of the font to BIRT config file fontsConfig.xml
Step 03: Restart of BIRT
Step 04: created test report and copied google test glyphs to the report
Step 05: testing of PDF, DOCX, XLSX, HTML (preview)

The results looks good the font with the glyphs are display.
(I used BIRT 12, JDK 11, Windows 10)

*01 Screen, fontsConfig.xml*
![BIRT-fontsConfig xml](https://user-images.githubusercontent.com/41593722/226734837-cb08a540-587c-42b8-8bf2-325eaab499ca.png)

*02 Screen, output formats HTML, DOCX, XLSX*
![BIRT_report_dev_font_NotoSansDevanagari-HTML-DOCX-XLSX](https://user-images.githubusercontent.com/41593722/226735028-930a32f5-6605-4fe9-955b-2f7549cb10d7.png)

*03 Example, summary of BIRT-report, fontsConfig.xml and output files*
[font_NotoSansDevanagari.zip](https://github.com/eclipse/birt/files/11033419/font_NotoSansDevanagari.zip)

I hope the screens and example will help you to solve your problem.

**Comment by Akshay-64** [comment]
thanks for reply
I am made all the changes as you mentioned above but except the step-4 Could you please elaborate the step-4(**copied google test glyphs to the report**)Please give me the brief about it

**Reply by speckyspooky** [reply]
On the google page was a test area to produce the glyphs. There I marked the glyphs like normal text and used realy only CTRL+C (browser) and CTRL+V on BIRT-text to copy the values.
I haven't any hint which key combination of the keyboard is to use to get the glyphs directly like standard text.

**Reply by Akshay-64** [reply]
I think step-4 was the mandatory is any other alternative step instead of step-4 because I am unable to achieve this step. Please share the step-4 text and also please share the screenshots from where you copied and paste the text in birt.
thanks once again  for your affect shown to solve my problem .

**Reply by speckyspooky** [reply]
Here is an text example which you can copy: चूंकि मानव परिवार के सभी सदस्यों के जन्मजात गौरव और समान
This text is also available on my zip file at the report if you show the xml source.

![report-text-example](https://user-images.githubusercontent.com/41593722/227027028-97c49775-3fcd-460c-bdbf-e822c8b9de1f.png)

An additional screen of my copy action. I don't know if this screen will help you really:

![text-copy](https://user-images.githubusercontent.com/41593722/227027204-438a7e90-a400-4ebd-bb86-2869c7c41a60.png)

**Reply by Akshay-64** [reply]
After doing reinstallation of eclipse I was able to achieve it . But for dynamic text and different languages like(japanese, chinese, bengali, hindi etc) I am unable to achieve it and is there any solution to achieve this problem .
Thanks for your suggestions to achieve static text for hindi. If possible try to help for dynamic text and for different languges.
Thanks once again

**Reply by hvbtup** [reply]
@Akshay-64  ... But what @speckyspooky showed you *is* dynamic text. Maybe your issue is caused by something else outside of BIRT, e.g. the storage of text in a database?

**Reply by Akshay-64** [reply]
In above example dynamic text was not mandatory so you skip it

**Comment by kiranduba** [comment]
Hi All, Thanks for your suggestions which helped me to reach up to getting font recognized by BIRT, but Im facing that its not getting rendered on generated PDF. Can you please help me here. Thanks. 

![image](https://github.com/eclipse-birt/birt/assets/17720995/c0fc2fdc-36b7-493c-98a7-de96891ae098)

**Comment by hvbtup** [comment]
You did not provide an example of your report for others to test.
In the screenshot from the designer, it seems that you did not specify a font name (it uses the default "Serif") - compare this with the example Thomas gave you.

And you did not say how the PDF was generated.
